docs/sdk/src/reference_docs/runtime_vs_smart_contract.rs

```diff
@@ -1,6 +1,73 @@
 //!
-//! Why one can be weighed, and one MUST be metered.
-//! https://forum.polkadot.network/t/where-contracts-fail-and-runtimes-chains-are-needed/4464/3
+//! Both Runtimes and Smart Contracts serve distinct purposes. Runtimes offer deep customization for
+//! blockchain development, while Smart Contracts provide a more accessible approach for
+//! decentralized applications. Understanding their differences is crucial in choosing the right
+//! approach for a specific solution.
+//!
+//! ## Substrate
+//! Substrate is a modular framework that enables the creation of purpose-specific blockchains. In
+//! the Polkadot ecosystem you can find two distinct approaches for on-chain code execution:
+//! [Runtime Development](#runtime-in-substrate) and [Smart Contracts](#smart-contracts).
+//!
+//! ## Smart Contracts
+//! Smart Contracts are autonomous, programmable constructs deployed on the blockchain.
+//! In [FRAME](frame), Smart Contracts capabilities can be realized through the
+//! [`pallet_contracts`](../../../pallet_contracts/index.html) for WASM-based contracts or the
+//! [`pallet_evm`](../../../pallet_evm/index.html) for EVM-compatible contracts. These pallets
+//! enable Smart Contract developers to build applications and systems on top of a Substrate-based
+//! blockchain.
+//!
+//! ## Runtime in Substrate
+//! The Runtime is the state transition function of a Substrate-based blockchain. It defines the
+//! rules for processing transactions and blocks, essentially governing the behavior and
+//! capabilities of a blockchain.
+//!
+//! ## Comparative Table
+//!
+//! | Aspect                | Runtime                                                                 | Smart Contracts                                                      |
+//! |-----------------------|-------------------------------------------------------------------------|----------------------------------------------------------------------|
+//! | **Design Philosophy** | Core logic of a blockchain, allowing broad and deep customization.      | Designed for Dapps on top of the the blockchain.|
+//! | **Development Complexity** | Requires in-depth knowledge of Rust and Substrate. Suitable for complex blockchain architectures.         | Easier to develop with knowledge of Smart Contract languages like Solidity or [ink!](https://use.ink/). |
+//! | **Upgradeability and Flexibility** | Offers seamless upgradeability, allowing entire blockchain logic modifications without hard forks. | Less flexible in upgrading but offers more straightforward deployment and iteration. |
+//! | **Performance and Efficiency** | More efficient, optimized for specific needs of the blockchain.        | Can be less efficient due to its generic nature (e.g. the overhead of a virtual machine).     |
+//! | **Security Considerations** | Security flaws can affect the entire blockchain.                        | Security risks usually localized to the individual contract.        |
```

Comment:
I think this last point is very important. Its not just security issues, also just normal bugs.  
A bug in a runtime will bring down the whole blockchain, whereas a bug in a contract just breaks the Dapp.  
I think this should be highlighted a bit more.